packages/frontend/src/features/sqlRunner/hooks/useSqlChartResults.ts

```diff
@@ -0,0 +1,153 @@
+                        // Enqueue the next data chunk into our target stream
+                        controller.enqueue(value);
+
+                        push();
+                    });
+                }
+
+                push();
+            },
+        });
+
+        const responseStream = new Response(stream, {
+            headers: { 'Content-Type': 'application/json' },
+        });
+        const result = await responseStream.text();
+
+        // Split the JSON strings by newline
+        const jsonStrings = result.trim().split('\n');
+        const jsonObjects: ResultRow[] = jsonStrings
+            .map((jsonString) => {
+                try {
+                    return JSON.parse(jsonString);
+                } catch (e) {
+                    throw new Error('Error parsing JSON');
+                }
+            })
+            .filter((obj) => obj !== null);
+
+        return jsonObjects;
+    } catch (e) {
+        // convert error to ApiError
+        throw <ApiError>{
+            status: 'error',
+            error: {
+                name: 'Error',
+                statusCode: 500,
+                message: e.message,
+                data: {},
+            },
+        };
+    }
+};
+
+const getSqlChartResults = async ({
+    projectUuid,
+    slug,
+}: {
+    projectUuid: string;
+    slug: string;
+}) => {
+    const scheduledJob = await lightdashApi<ApiJobScheduledResponse['results']>(
+        {
+            url: `/projects/${projectUuid}/sqlRunner/saved/slug/${slug}/results-job`,
+            method: 'GET',
+            body: undefined,
+        },
+    );
+
+    // recursively check job status until it's complete
+    async function getCompleteJob(): Promise<
+        ApiSqlRunnerJobStatusResponse['results']
+    > {
+        const job = await getSchedulerJobStatus<
+            ApiSqlRunnerJobStatusResponse['results']
+        >(scheduledJob.jobId);
+        if (
+            job.status === SchedulerJobStatus.SCHEDULED ||
+            job.status === SchedulerJobStatus.STARTED
+        ) {
+            return new Promise((resolve) => {
+                setTimeout(async () => {
+                    resolve(await getCompleteJob());
+                }, 3000);
+            });
+        }
+        if (job.status === SchedulerJobStatus.COMPLETED) {
+            return job;
+        } else {
+            throw <ApiError>{
+                status: 'error',
+                error: {
+                    name: 'Error',
+                    statusCode: 500,
+                    message: isErrorDetails(job.details)
+                        ? job.details.error
+                        : 'Job failed',
+                    data: {},
+                },
+            };
+        }
+    }
+
+    const job = await getCompleteJob();
+    const url =
+        job?.details && !isErrorDetails(job.details)
+            ? job.details.fileUrl
+            : undefined;
+    return getResultsStream(url);
+};
+
+export const useSqlQueryRun = (
```

Comment:
It may be confusing to have two `useSqlQueryRun` hooks (on in `useSqlChartResults.tsx` and another in `useSqlQueryRun.tsx`)
Maybe this should be named `useSqlChartResults`?